### PR TITLE
Simplify derivative

### PR DESCRIPTION
**Pull request by isuruf**
This assumes that `Derivative(f, x, y)` is the same as `Derivative(f, y, x)`

### CODE REVIEW

**Comment by cbehan** [comment]
I'm wondering if I can help move this along. Here's a patch, https://paste.debian.net/310053/ which fixes the one test that was still failing. It probably wouldn't make sense to file a pull request for a pull request.

**Comment by isuruf** [comment]
@cbehan, I fixed the failing test in f7fd013. 
>It probably wouldn't make sense to file a pull request for a pull request.
Actually, when somebody wants to add to a pull request, that's what we do. (Send a PR to the original author's fork)

@certik, now this is ready. Let me know if the assumption that `Derivative(f, x, y)` is the same as `Derivative(f, y, x)` is okay.

File: symengine/functions.cpp [hunk 1]
```diff
@@ -1551,6 +1552,8 @@ RCP<const Basic> FunctionWrapper::diff(const RCP<const Symbol> &x) const
 Derivative::Derivative(const RCP<const Basic> &arg, const vec_basic &x)
     : arg_{arg}, x_{x}
 {
+    // TODO: use a multi-set for x_ instead of vector
+    std::sort(x_.begin(), x_.end(), RCPBasicKeyLessCmp());
```
Comment (certik) [inline symengine/functions.cpp]: You don't want to be doing any work in the constructor. All work must happen before you call the constructor, and then you just check this in `is_canonical`. That way you can quickly construct derivatives if you already have the data in the proper form.

File: symengine/tests/basic/test_functions.cpp [hunk 2]
```diff
@@ -747,8 +747,8 @@ TEST_CASE("Derivative: functions", "[functions]")
 
     f = function_symbol("f", {x, y});
     r1 = f->diff(x)->diff(y);
-    r2 = Derivative::create(f, {x, y});
-    r3 = Derivative::create(f, {y, x});
+    r2 = f->diff(y)->diff(x);
+    r3 = f->diff(x)->diff(z);
```
Comment (certik) [inline symengine/tests/basic/test_functions.cpp]: We should keep the tests for `create`, shouldn't we? And just add more tests for the `diff`.

**Comment by certik** [comment]
I left some comments.

File: symengine/tests/basic/test_functions.cpp (hunk 2 above)
Comment (isuruf) [inline symengine/tests/basic/test_functions.cpp]: I added more create tests below

**Comment by isuruf** [comment]
Used `std::multiset`. Advantage of this is that we can simply do `Derivative::create(f, {x,y})` without worrying about the fact that `x, y` is in the correct order or not which is platform dependent. See https://github.com/isuruf/symengine/pull/6

**Comment by cbehan** [comment]
Thanks for supporting `std::multiset` so quickly! I just filed a pull for this pull which makes `Subs::subs` start off the same way as most other `subs` methods. There's one thing I need to say about this chain rule support: I just realized that I don't need it right now. I still think it would be good to have eventually, but if you have other priorities, take your time.

**Comment by cbehan** [comment]
Is there a problem with https://github.com/isuruf/symengine/pull/7? That was the last change I could think of but it's up to you whether it goes in with this pull request or separately.

**Comment by rwst** [comment]
LGTM. Thanks!

**Pull request merged**